publication-commons/src/main/java/no/unit/nva/publication/rightsretention/RightsRetentionsValueFinder.java

```diff
@@ -1,126 +0,0 @@
-        throws BadRequestException, UnauthorizedException {
-        var fileRightsRetention = file.getRightsRetentionStrategy();
-
-        return rrsIsIrrelevant(file, resource)
-                   ? NullRightsRetentionStrategy.create(configuredRightsRetention)
-                   : getRrsForUnPublishedFile(file, fileRightsRetention);
-    }
-
-    private boolean rrsIsIrrelevant(File file, Resource resource) {
-        return !PublisherVersion.ACCEPTED_VERSION.equals(file.getPublisherVersion())
-               || file instanceof InternalFile
-               || !isAcademicArticle(resource);
-    }
-
-    private static boolean isAcademicArticle(Resource resource) {
-        return Optional.ofNullable(resource)
-                   .map(Resource::getEntityDescription)
-                   .map(EntityDescription::getReference)
-                   .map(Reference::getPublicationInstance)
-                   .map(instance -> instance instanceof AcademicArticle)
-                   .orElse(false);
-    }
-
-    @JacocoGenerated
-    private static Set<RightsRetentionStrategyConfiguration> getAllowedConfigurations(RightsRetentionStrategy rrs) {
-        return switch (rrs) {
-            case OverriddenRightsRetentionStrategy strategy -> Set.of(OVERRIDABLE_RIGHTS_RETENTION_STRATEGY);
-            case NullRightsRetentionStrategy strategy -> Set.of(NULL_RIGHTS_RETENTION_STRATEGY);
-            case CustomerRightsRetentionStrategy strategy ->
-                Set.of(RIGHTS_RETENTION_STRATEGY, OVERRIDABLE_RIGHTS_RETENTION_STRATEGY);
-            case FunderRightsRetentionStrategy strategy ->
-                Set.of(NULL_RIGHTS_RETENTION_STRATEGY, OVERRIDABLE_RIGHTS_RETENTION_STRATEGY);
-            default -> throw new IllegalArgumentException("Unknown rrs type " + rrs);
-        };
-    }
-
-    @JacocoGenerated
-    private static RightsRetentionStrategyConfiguration getConfigFromCustomerDto(CustomerApiRightsRetention config) {
-        return switch (config.getType()) {
-            case "NullRightsRetentionStrategy" -> NULL_RIGHTS_RETENTION_STRATEGY;
-            case "RightsRetentionStrategy" -> RIGHTS_RETENTION_STRATEGY;
-            case "OverridableRightsRetentionStrategy" -> OVERRIDABLE_RIGHTS_RETENTION_STRATEGY;
-            default -> throw new IllegalArgumentException("Unknown RightsRetentionStrategy type " + config.getType());
-        };
-    }
-
-    private boolean isValid(RightsRetentionStrategy rrs) {
-        return getAllowedConfigurations(rrs).contains(rrs.getConfiguredType())
-               || isAllowedToOverrideRrs(rrs, permissionStrategy);
-    }
-
-    private boolean isAllowedToOverrideRrs(RightsRetentionStrategy rrs,
-                                           PublicationPermissions permissionStrategy) {
-        if (nonNull(permissionStrategy) && rrs instanceof OverriddenRightsRetentionStrategy) {
-            return permissionStrategy.isPublishingCuratorOnPublication();
-        }
```

Comment:
Bug is here -> isPublishingCuratorOnPublication() is used only here and is based on deprecated allowed operation. 
Impossible to fix this without large refactoring, cause this check is based on PublicationPermissions, but new one on FilePermission and FileEntry.